src/kOS/Utilities/PartUtilities.cs

```diff
@@ -1,12 +1,23 @@
-﻿using System;
+﻿using kOS.Safe.Encapsulation;
+using System;
 
 namespace kOS.Utilities
 {
     public static class PartUtilities
     {
         public static float CalculateCurrentMass(this Part part)
         {
-            return part.HasPhysics() ? part.mass + part.GetResourceMass() : 0;
+            if (part.HasPhysics())
```

Comment:
We could substitute it with `part.HasPhysics() && part.rb != null ? part.rt.mass : 0` if I'm not mistaken.